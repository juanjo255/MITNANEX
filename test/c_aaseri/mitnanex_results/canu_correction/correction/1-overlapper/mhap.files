1-overlapper/results/000001.mhap
1-overlapper/queries
1-overlapper/queries/000001
