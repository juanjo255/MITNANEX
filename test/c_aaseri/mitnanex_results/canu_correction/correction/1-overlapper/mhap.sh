#!/bin/sh


#  Path to Canu.

bin="/Users/jjpc/miniforge3/envs/mitnanex/bin"

#  Report paths.

echo ""
echo "Found perl:"
echo "  " `which perl`
echo "  " `perl --version | grep version`
echo ""
echo "Found java:"
echo "  " `which /Users/jjpc/miniforge3/envs/mitnanex/lib/jvm/bin/java`
echo "  " `/Users/jjpc/miniforge3/envs/mitnanex/lib/jvm/bin/java -showversion 2>&1 | head -n 1`
echo ""
echo "Found canu:"
echo "  " $bin/canu
echo "  " `$bin/canu -version`
echo ""


#  Environment for any object storage.

export CANU_OBJECT_STORE_CLIENT=
export CANU_OBJECT_STORE_CLIENT_UA=
export CANU_OBJECT_STORE_CLIENT_DA=
export CANU_OBJECT_STORE_NAMESPACE=
export CANU_OBJECT_STORE_PROJECT=




#  Discover the job ID to run, from either a grid environment variable and a
#  command line offset, or directly from the command line.
#
if [ x$CANU_LOCAL_JOB_ID = x -o x$CANU_LOCAL_JOB_ID = xundefined -o x$CANU_LOCAL_JOB_ID = x0 ]; then
  baseid=$1
  offset=0
else
  baseid=$CANU_LOCAL_JOB_ID
  offset=$1
fi
if [ x$offset = x ]; then
  offset=0
fi
if [ x$baseid = x ]; then
  echo Error: I need CANU_LOCAL_JOB_ID set, or a job index on the command line.
  exit
fi
jobid=`expr -- $baseid + $offset`
if [ x$baseid = x0 ]; then
  echo Error: jobid 0 is invalid\; I need CANU_LOCAL_JOB_ID set, or a job index on the command line.
  exit
fi
if [ x$CANU_LOCAL_JOB_ID = x ]; then
  echo Running job $jobid based on command line options.
else
  echo Running job $jobid based on CANU_LOCAL_JOB_ID=$CANU_LOCAL_JOB_ID and offset=$offset.
fi

if [ $jobid -eq 1 ] ; then
  blk="000001"
  slf=""
  qry="000001"
fi


if [ x$qry = x ]; then
  echo Error: Job index out of range.
  exit 1
fi

if [ -e ./results/$qry.ovb ]; then
  echo Job previously completed successfully.
  exit
fi


if [ -e ./queries.tar -a ! -d ./queries ] ; then
  tar -xf ./queries.tar
fi


if [ ! -d ./results ]; then  mkdir -p ./results;  fi
if [ ! -d ./blocks  ]; then  mkdir -p ./blocks;   fi

for ii in `ls ./queries/$qry` ; do
  echo Fetch blocks/$ii
done

echo ""
echo Running block $blk in query $qry
echo ""

if [ ! -e ./results/$qry.mhap.ovb ] ; then
  #  Make a fifo so we can check return status on both
  #  mhap and mhapConvert, and still pipe results so we
  #  stop filling up disks.
  rm -f  $qry-pipe
  mkfifo $qry-pipe

  #  Start up the consumer.
  $bin/mhapConvert \
    -S ../../c_aaseri_ERR10466724_collected_reads.seqStore \
    -o ./results/$qry.mhap.ovb.WORKING \
    -minlength 500 \
    $qry-pipe \
  && \
  touch ./results/$qry.mcvt.success &

  #  Start up the producer.
  /Users/jjpc/miniforge3/envs/mitnanex/lib/jvm/bin/java  -XX:ParallelGCThreads=8 -server -Xms5530m -Xmx5530m \
    -jar  $bin/../share/java/classes/mhap-2.1.3.jar  \
    --repeat-weight 0.9 --repeat-idf-scale 10 -k 16 \
    --store-full-id \
    --num-hashes 256 \
    --num-min-matches 3 \
    --threshold 0.8 \
    --filter-threshold 0.0000001 \
    --ordered-sketch-size 1000 \
    --ordered-kmer-size 14 \
    --min-olap-length 500 \
    --num-threads 8 \
    -s  ./blocks/$blk.dat $slf  \
    -q  queries/$qry  \
  > $qry-pipe \
  && \
  touch ./results/$qry.mhap.success

  #  Wait for mhapConvert to finish.
  wait

  #  Now that they're done, check status.
  if [ -e ./results/$qry.mhap.success -a -e ./results/$qry.mcvt.success ] ; then
    mv ./results/$qry.mhap.ovb.WORKING ./results/$qry.mhap.ovb
    rm -f ./results/$qry.mhap.success
    rm -f ./results/$qry.mcvt.success
  fi

  #  And destroy the pipe.
  rm -f  $qry-pipe
fi

if [ -e ./results/$qry.mhap.ovb ] ; then
  mv -f ./results/$qry.mhap.ovb ./results/$qry.ovb
fi


exit 0
