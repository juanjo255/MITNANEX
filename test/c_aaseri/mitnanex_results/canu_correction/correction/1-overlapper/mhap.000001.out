
Found perl:
   /Users/jjpc/miniforge3/envs/mitnanex/bin/perl
   This is perl 5, version 32, subversion 1 (v5.32.1) built for darwin-thread-multi-2level

Found java:
   /Users/jjpc/miniforge3/envs/mitnanex/lib/jvm/bin/java
   openjdk version "21.0.1" 2023-10-17 LTS

Found canu:
   /Users/jjpc/miniforge3/envs/mitnanex/bin/canu
   canu 2.2

Running job 1 based on command line options.

Running block 000001 in query 000001

Running with these settings:
--filter-threshold = 1.0E-7
--help = false
--max-shift = 0.2
--min-olap-length = 500
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 256
--num-min-matches = 3
--num-threads = 8
--ordered-kmer-size = 14
--ordered-sketch-size = 1000
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = true
--supress-noise = 0
--threshold = 0.8
--version = false
-f = 
-h = false
-k = 16
-p = 
-q = queries/000001
-s = ./blocks/000001.dat

Processing files for storage in reverse index...
Current # sequences loaded and processed from file: 5000...
Current # sequences stored: 5000...
Stored 5864 sequences in the index.
Processed 5864 unique sequences (fwd and rev).
Time (s) to read and hash from file: 0.546183375
Time (s) to score and output to self: 10.875357541000001
Total scoring time (s): 10.87867025
Total time (s): 11.425470709
MinHash search time (s): 3.4070041900000003
Total matches found: 698857
Average number of matches per lookup: 238.3550477489768
Average number of table elements processed per lookup: 14066.597544338336
Average number of table elements processed per match: 59.015312145403136
Average % of hashed sequences hit per lookup: 22.122969435443494
Average % of hashed sequences hit that are matches: 18.373291808383353
Average % of hashed sequences fully compared that are matches: 95.57959894854604
